global/test/TestCommandLineArgumentsTutorial.hpp

```diff
@@ -0,0 +1,173 @@
+    
+    /*
+      In addition to using unsigned integers as command line arguments we can also pass in both doubles and a vector of arguments.
+      This could be very useful for several reasons, such as having a large list of parameters one may wish to pass into a test. 
+
+      For this test change the runcommandlinetutorial.sh bash script from line 16 onwards as follow by copy and pasting between START and END.
+      **START**
+      # Here we will declare some values we wish to later pass to a for loop.
+      # As bash script cannot directly handle double arithmetic we will First
+      # declare these as larger variables to be handled by another programme later.
+      N=2000
+      L=3000
+      M=4000
+
+      # Here we set up a simple for loop over variables i,j and k based on the values of N,L and M.
+      for ((i = 1; i <= N; i += 1000)); do
+       for ((j = 1001; j <= L; j += 1000)); do
+        for ((k = 2001; k <= M; k += 1000)); do
+          # As bash cannot directly handle double floating point arithmetic we will utilise
+          # awk to divide our varaibles and convert them to doubles.
+           idouble=$(awk "BEGIN {printf \"%.2f\",$i/120}")
+           jdouble=$(awk "BEGIN {printf \"%.2f\",$j/150}")
+           kdouble=$(awk "BEGIN {printf \"%.2f\",$k/180}")
+
+           # Each loop runs an instance of the TestCommandLineArgumentsTutorial with a vector
+           # containing the double corrected version of our varaibles.
+           ~/build/global/test/TestCommandLineArgumentsTutorial --my-vector-of-arguments $idouble $jdouble $kdouble &
+        done
+       done
+      done
+      **END**
+    */
+    void TestCommandLineDoubleTutorial()
+    {  
+        // Here an if statement is utilised to check that we have passed some command line arguments to our test.
+        // To ensure the test can still run in the absence of command line arguments we will initialise the vector 
+        // with some default values if no command line option exists.
+        if (CommandLineArguments::Instance()->OptionExists("--my-vector-of-arguments"))
+        { 
+           // First, we need to setup our vector based on the vector passed in as our command line arguement. 
+           std::vector<double> vector_of_doubles = CommandLineArguments::Instance()->GetDoublesCorrespondingToOption("--my-vector-of-arguments");
+           double sum = 0.0;
+           // Now summing over our vector.
+           for(double i = 0; i < vector_of_doubles.size(); i++){
+            sum += vector_of_doubles[i]; 
+           }
+           // We will finally cout our result for the summed vector components.
+           // Here we have chosen to set our precision to 4 to ensure the correct number of significant figures.
+           std::cout << std::setprecision(5) << "When we add "<< vector_of_doubles[0] << ", " << vector_of_doubles[1] << " and " << vector_of_doubles[2] << " together from our vector we get " << sum  <<"\n"; 
+
+        } else {
```

Comment:
Can we just remove the else block, it is not doing anything here?

Reply:
It could be removed but I think I could also add some basic testing style structure here ? As other tutorials seem to have some small additions for testing purposes. I suppose it could also act as a way to ensure nothing odd happens. I'm also with for removing it though.